pkg/metadata_test.go

```diff
@@ -106,7 +106,7 @@ var metadataMarshalData = map[string]marshalData{
 		},
 	},
 	"federation + rp": {
-		Data: []byte(`{"openid_relying_party":{"application_type":"web","client_name":"test rp","client_registration_types":["automatic"],"client_uri":"https://rp.example.org","contacts":["contact@rp.example.org"],"grant_types":["authorization_code"],"jwks_uri":"https://rp.example.org/jwks","organization_name":"test rp org","policy_uri":"https://rp.example.org/policy","redirect_uris":["https://rp.example.org/redirect"],"response_types":["code"],"scope":"openid profile email","tos_uri":"https://rp.example.org/tos"},"federation_entity":{"contacts":["contact@rp.example.org"],"organization_name":"test rp org"}}`),
+		Data: []byte(`{"federation_entity":{"contacts":["contact@rp.example.org"],"organization_name":"test rp org"},"openid_relying_party":{"application_type":"web","client_name":"test rp","client_registration_types":["automatic"],"client_uri":"https://rp.example.org","contacts":["contact@rp.example.org"],"grant_types":["authorization_code"],"jwks_uri":"https://rp.example.org/jwks","organization_name":"test rp org","policy_uri":"https://rp.example.org/policy","redirect_uris":["https://rp.example.org/redirect"],"response_types":["code"],"scope":"openid profile email","tos_uri":"https://rp.example.org/tos"}}`),
```

Comment:
Providing `MarshalJSON` and `UnmarshalJSON` implementations for `Metadata` seems to have changed the order in which JSON keys get written, so I had to change this test data to match. AFAICT there's no actual change to the JSON object.